src/operators/transfermatrix.jl

```diff
@@ -0,0 +1,250 @@
+
+VectorInterface.scalartype(::Type{InfiniteTransferPEPS{T}}) where {T} = scalartype(T)
+
+function virtual_space(O::InfiniteTransferPEPS, i, dir)
+    return prod([space(O.top[i], dir + 1), space(O.bot[i], dir + 1)'])
+end
+
+"""
+    const MultilineTransferPEPS = MPSKit.Multiline{<:InfiniteTransferPEPS}
+
+Type that represents a multi-line transfer operator, where each line each corresponds to a
+row of a partition function encoding the overlap between 'ket' and 'bra' `InfinitePEPS`
+states.
+"""
+const MultilineTransferPEPS = MPSKit.Multiline{<:InfiniteTransferPEPS}
+Base.convert(::Type{MultilineTransferPEPS}, O::InfiniteTransferPEPS) = MPSKit.Multiline([O])
+
+"""
+    MultilineTransferPEPS(T::InfinitePEPS, dir)
+
+Construct a multi-row transfer operator corresponding to the partition function representing
+the norm of the state `T`. The partition function is first rotated such
+that the direction `dir` faces north.
+"""
+function MultilineTransferPEPS(T::InfinitePEPS, dir)
+    rowsize = size(T, mod1(dir, 2))  # depends on dir
+    return MPSKit.Multiline(map(cr -> InfiniteTransferPEPS(T, dir, cr), 1:rowsize))
+end
+
+#
+# PEPO
+#
+
+"""
+    InfiniteTransferPEPO{T,O}
+
+Represents an infinite transfer operator corresponding to a single row of a partition
+function which corresponds to the expectation value of an `InfinitePEPO` between 'ket' and
+'bra' `InfinitePEPS` states.
+"""
+struct InfiniteTransferPEPO{T,O}
+    top::PeriodicArray{T,1}
+    mid::PeriodicArray{O,2}
+    bot::PeriodicArray{T,1}
+    function InfiniteTransferPEPO(
+        top::PeriodicArray{T,1}, mid::PeriodicArray{O,2}, bot::PeriodicArray{T,1}
+    ) where {T,O}
+        size(top, 1) == size(bot, 1) == size(mid, 1) ||
+            throw(ArgumentError("Top PEPS, bottom PEPS and PEPO rows should have length"))
+        return new{T,O}(top, mid, bot)
+    end
+end
+
+InfiniteTransferPEPO(top, mid) = InfiniteTransferPEPO(top, mid, top)
+
+"""
+    InfiniteTransferPEPO(T::InfinitePEPS, O::InfinitePEPO, dir, row)
+
+Constructs a transfer operator corresponding to a single row of a partition function
+representing the expectation value of `O` for the state `T`. The partition function is first
+rotated such that the direction `dir` faces north, after which its `row`th row from the
+north is selected.
+"""
+function InfiniteTransferPEPO(T::InfinitePEPS, O::InfinitePEPO, dir, row)
+    T = rotate_north(T, dir)
+    O = rotate_north(O, dir)
+    return InfiniteTransferPEPO(PeriodicArray(T[row, :]), PeriodicArray(O[row, :, :]))
+end
+
+Base.size(transfer::InfiniteTransferPEPO) = size(transfer.top)
+Base.size(transfer::InfiniteTransferPEPO, args...) = size(transfer.top, args...)
+Base.length(transfer::InfiniteTransferPEPO) = size(transfer, 1)
+height(transfer::InfiniteTransferPEPO) = size(transfer.mid, 2)
+Base.getindex(O::InfiniteTransferPEPO, i) = (O.top[i], O.bot[i], Tuple(O.mid[i, :]))
+
+Base.iterate(O::InfiniteTransferPEPO, i=1) = i > length(O) ? nothing : (O[i], i + 1)
```

Comment:
```suggestion
Base.iterate(O::InfiniteTransferPEPO, args...) = iterate(zip(O.top, O.bot, eachcol(O.mid)), args...)
```
(just a suggestion, can safely be ignored)